libres/tests/tmpdir.cpp

```diff
@@ -0,0 +1,127 @@
+#include <string>
+#include <optional>
+#include <regex>
+
+#include <unistd.h>
+#include <pwd.h>
+
+#include <catch2/catch.hpp>
+#include <fmt/format.h>
+
+#include "tmpdir.hpp"
+
+using namespace std::string_literals;
+namespace fs = std::filesystem;
+
+namespace {
+/**
+ * Convert string into a nicer directory name
+ *
+ * Downcases, replaces whitespace-ish chars with '_' and removes the
+ * rest.
+ */
+std::string make_dirname(const std::string &dirname) {
+    std::string out;
+    out.reserve(dirname.size());
+
+    for (char c : dirname) {
+        if ('a' <= c && c <= 'z')
+            out.push_back(c);
+        else if ('A' <= c && c <= 'Z')
+            out.push_back(c + 'a' - 'A');
+        else if (c == '_' || c == '-' || c == ' ')
+            out.push_back('_');
+    }
+    return out;
+}
+
+/**
+ * Get POSIX username of the current user
+ *
+ * Falls back to the user's UID if their username is unavailable.
+ */
+std::string get_username() {
+    auto uid = getuid();
+    auto pw = getpwuid(uid);
+    return pw ? std::string{pw->pw_name} : std::to_string(uid);
+}
+
+/**
+ * Create a new directory suffixed with a number
+ *
+ * The prefix is normalised with 'make_dirname'.
+ *
+ * \param basepath Directory in which to create the new directory
+ * \param prefix Prefix of the directory to be created
+ */
+std::filesystem::path make_numbered_dir(const fs::path &basepath,
+                                        const std::string &prefix) {
+    auto norm_prefix = make_dirname(prefix);
```

Comment:
`make_dirname(prefix)` does not read great IMO - perhaps rename `make_dirname` to `clean_string` or something?
Any good reason to care about the naming of temp-dirs?

Reply:
You're right.

We don't really, but the test names are arbitrary and may contain spaces, numbers and `/` (the only illegal character in a Linux filesystem node). Other than the latter, which will have odd behaviour, the rest make for awkwardly-named directories: `/tmp/catch2-of-zom/ert_test_suite/Multiply two matrices A & B0`.